src/utils/client.js

```diff
@@ -215,14 +220,19 @@ export default class Client {
     });
   }
 
-  submitCustom(customKeyValues) {
+  submitCustom(customKeyValues, signatureRef) {
     const { userKey } = initInfo();
     const url = `${API_URL}panel/custom/submit/v2/`;
 
     const jsonKeyVals = {};
     const formData = new FormData();
     Object.entries(customKeyValues).forEach(([key, val]) => {
-      if (val instanceof Blob) {
+      if (isBase64PNG(val)) {
+        if (signatureRef[key]) signatureRef[key].fromDataURL(val);
```

Comment:
I think this line can be folded into the original if block, and the conditional can become `isBase64PNG(val) || val instanceof Blob`?